### PR TITLE
Add League Endpoint

### PR DESCRIPTION
**Pull request by Skylude**
Add League Endpoint

**Why**
We want a league endpoint in order to display all leagues that exist.

**How**
Add nuke-migrations to makefile to drop and recreate migrations. Update README.md to have instructions on make commands and how to add an endpoint. Add db methods to fetch leagues. Add handler to return all leagues. Add docs to retrieve leagues.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,6 +2,50 @@
...
+1. Navigate to the server directory: \
+    `backend/internal/server/apis`
+
+### Add DB methods used in API endpoint
+1. Navigate to the db directory: \
+    `backend/internal/db`
+
+### Update Open API docs**
+1. Navigate to the open api spec directory: \
+   `static/oas/v1`
```
Comment (ryanbarlow1) [inline README.md:51]: This is great for the README, but it's all backend focused. Should we create a README in the backend folder? Or create a stubbed frontend section above it?

File: backend/internal/server/apis/league/league.go [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+func init() {
+	apis.RegisterHandler(fiber.MethodGet, "/leagues", auth.Public, getLeaguesHandler)
+}
+
+func getLeaguesHandler(c *fiber.Ctx) error {
+	db := db.GetSession(c)
+	leagues, err := db.GetLeagues()
+	if err != nil {
+		// todo: create ticket to standardize this error message and pass in model name
+		log.WithErr(err).Alert("Failed to get all leagues from the database")
```
Comment (ryanbarlow1) [inline backend/internal/server/apis/league/league.go:22]: I don't believe this works unless you add `log := locals.Logger(c)` to the method

File: backend/internal/server/apis/league/league.go [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+	jsonData, err := json.Marshal(leagues)
+	if err != nil {
+		// todo: create ticket to standardize this error message and pass in model name
+		log.WithErr(err).Alert("Failed to serialize leagues response payload")
+		return err
+	}
+
+	c.Type("json")
+
+	return c.Send(jsonData)
```
Comment (ryanbarlow1) [inline backend/internal/server/apis/league/league.go]: [Here's an example](https://github.com/tannerwheeler/powerplay-backend-stats/blob/penalty-improvements/backend/internal/server/apis/stats/penalty.go) of how to do this using the `responder.OkWithData()` that Dr. Christensen mentioned we should be using, I just updated the Penalty API in a branch yesterday.

**Review by ryanbarlow1 — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (Skylude) [reply README.md:51]: Yes I was already thinking this! I'll update in another PR for FE, scheduling and the raspberry Pi!

Comment (tannerwheeler) [reply README.md:51]: I have created a new issue (#120) for this addition. Please let me know if there are changes that need to be made to the issue. I do not think I captured the intent correctly.

File: backend/internal/server/apis/league/league.go (hunk 2 above)
Comment (tannerwheeler) [reply backend/internal/server/apis/league/league.go:22]: I think Ryan is correct. I remember from class on 5/23 Dr. Christensen mentioning `locals.Logger(c)` will need to be used in every Handler method.

**Pull request merged**